### ISSUE TITLE
Linked block with multiple product might not get right ratio ?

### ISSUE THREAD

**Issue by Paraplegix** [Helfima/helmod#537]
Tested using Helmod 1.0.11 (pulled from repo)
With 10 Steel plate requested from Parent Block (PB) and the following setup of the linked Sub Block (SB): 
![image](https://github.com/Helfima/helmod/assets/12891208/88db9453-9ace-4324-8b63-9606ce08d7fa)
(Note: the block IS linked, but the button doesn't show as "on" in SB no matter where I toggle it, but still show in "parent view", another bug ?)
In there, only the steel in PB is set.

1) If I set the iron to any amount in SB, the steel becomes 0 both in PB and SB. For example I set 5 Iron : 
![image](https://github.com/Helfima/helmod/assets/12891208/c6f9d2d6-9757-4d40-ba1e-3f34be3792b5)
![image](https://github.com/Helfima/helmod/assets/12891208/6a59db56-548d-447c-823f-adb9899e3331)

2) If I set iron to any amount in PB (from start so no iron in SB), the steel stay at 10 and iron stay at 0:
![image](https://github.com/Helfima/helmod/assets/12891208/4d789cb0-cdbe-42cc-bd73-909080e18285)

3) If I set iron in both PB and SB, steel becomes 0 and iron changes in both parent and sb as requested. For example 10 iron in PB and 5 in SB
![image](https://github.com/Helfima/helmod/assets/12891208/1fb2dfe8-f692-40ce-9f2a-2353ab46f3a7)

4) If I set iron in PB, and steel and iron in SB, iron will be shown correctly in PB but will also show steel. For example 10 iron in PB, 10 steel and 5 iron in SB will show 20 steel and 10 iron in SB:
![image](https://github.com/Helfima/helmod/assets/12891208/00f50187-16ab-4ba0-ae36-a3109bd34bc3)

5) Similarly, if I set iron in both PB and SB, and Steel in both PB and SB, then all products can be requested from parent, but on the exact ratio set in SB. For example, with 40 steel and 10 iron in PB, 10 steel and 5 iron in SB, it will show correct value in SB, but 40 steel and 20 iron ins PB : 
![image](https://github.com/Helfima/helmod/assets/12891208/afbbf811-d08a-4612-9ba2-016828bfd8b9)

Imho linked block should completely override values inside blocks.

**Comment by Helfima** [comment]
you must see a sub-block same as recipe for the parent block
if you not set something the solver take a value
if you set a value you must set other value if necessary

**Comment by Paraplegix** [comment]
Yes but sometimes the "requests" are not set by me but by other recipes.

For example if I have a line dedicated to belt crafting. You have yellow belt which is iron plate and gear, and yellow underground belt is yellow belt + iron plate.

If I want to keep iron plates and gears in a linked sub block because that's "base material", that oblige me to manually maintain the correct input in the sub block. I honestly don't think that's user-friendly.
